docs/FAQ.md

```diff
@@ -298,14 +298,180 @@ jj config set --user git.private-commits "'''description(glob:'private:*')'''"
+this is what the evolog might look like after you made two edits to the same
+commit:
+
+```console
+$ # Note the "hidden", the commit ids on the right, and the unchanging change id on the left
+$ jj evolog
+@  lnvvtrzo jjfan@example.org 2025-02-28 21:01:10 31a347e0
+│  featureA
+○  lnvvtrzo hidden jjfan@example.org 2025-02-28 21:00:51 b8004ab8
+│  featureA
+○  lnvvtrzo hidden jjfan@example.org 2025-02-28 20:50:05 e4d831d
+   (no description set)
+```
+
+Since commit `b800` is hidden, it is considered obsolete and `jj log` (without
+arguments) will not show it, nor can it be accessed by its change id. However,
+most `jj` operations will work on it if you refer to it by its commit id.
+
+To find out which of these versions is the last time before we started working
+on feature B (the point where we should have created a new revision, but failed
+to do so), we can look at the actual changes between the `evolog` commits by
+running `jj evolog -p`:
+
+```console
+$ # When was the last saved point before we started working on feature B?
+$ jj evolog -p --git  # We use `--git` to make diffs clear without colors
+@  lnvvtrzo jjfan@example.org 2025-02-28 21:01:10 31a347e0
+│  featureA
+│  diff --git a/file b/file
+│  index 2b455c4207..2a7e05a01a 100644
+│  --- a/file
+│  +++ b/file
+│  @@ -1,1 +1,2 @@
+│   Done with feature A
+│  +Working on feature B
+○  lnvvtrzo hidden jjfan@example.org 2025-02-28 21:00:51 b8004ab8
+│  featureA
+│  diff --git a/file b/file
+│  index cb61245109..2b455c4207
+│  --- a/file
+│  +++ b/file
+│  @@ -1,1 +1,1 @@
+│  -Working on feature A
+│  +Done with feature A
+○  lnvvtrzo hidden jjfan@example.org 2025-02-28 20:50:05 e4d831d
+   (no description set)
+   diff --git a/file b/file
+   index 0000000000..cb61245109
+   --- /dev/null
+   +++ b/file
+   @@ 0,0 +1,1 @@
+   +Working on feature A
+...
+```
+
+The version of the revision when we were actually done with "featureA" is thus
+saved in the commit with id `b800`. The change after that belongs in a new
+revision.
+
+#### Step 2: Restore the existing revision to an older state and create a new revision with the current state
+
+The "featureA" revision is currently at commit `31a`:
+
+```console
+$ jj log
+@  lnvvtrzo jjfan@example.org 2025-02-28 21:01:10 31a347e0
+│  featureA
+◆  zzzzzzzz root() 00000000
+```
+
+We'd like to restore the "featureA" revision to its state at commit `b80` (see
+step 1 above for how we found that commit id), and to create a new revision
+called "featureB" that has the same contents as commit `31a`.
+
+First, we create a new empty child commit. Since it is empty, it has the same
+contents as `31a`.
+
+```console
+$ jj new -m "featureB"
+Working copy now at: pvnrkl 47171aa (empty) featureB
+Parent commit      : lnvvtr 31a347e featureA
+$ cat file
+Done with feature A
+Working on feature B
+```
+
+Now, we `jj restore` the revision `lnvvtr` to its state at commit `b80`. We use
```

Comment:
The glossary says that revision and commit are synonyms. 
Should this revision be "change"?

Reply:
I'm not immediately sure what's best to do here. "Change" sounds too generic, but I'm not sure where the consensus on terminology is. I wonder what people think; in particular if others feel that "change" would work OK for now everywhere I used "revision", that'd be helpful to know.

I hope we'll find something that works, even if temporarily, and isn't blocked fully resolving the terminology jungle. From this perspective, "change" seems promising.

Reply:
Isn't the `lnvvtr` a change ID?

Reply:
When we had a big discussion about these concepts on Discord, @arxanas broke my faith in the primary of “change” as a concept by pointing out that divergent change IDs break a lot of natural talk of “changes”. In this case, there could be another `lnvvtr` that `jj restore` on `@-` wouldn’t do anything to; it’s actually `31a347e` we’re evolving, not `lnvvtr`. I don’t know what implications that has for this FAQ, but maybe it’s reasonable to say “revision” here because `lnvvtr` is really just acting as a shorthand name for `31a347e` rather than being something we’re addressing directly.

Reply:
I tried writing it with "change" as the primary concept, and it seems to work OK. As Joy mentioned, it matches our definitions in the glossary.

I think it might be nice to clean up the terminology, but then I guess we should change this text together with the glossary.